main.tf

```diff
@@ -112,14 +112,24 @@ resource "aws_cloudfront_distribution" "this" {
       default_ttl = lookup(i.value, "default_ttl", null)
       max_ttl     = lookup(i.value, "max_ttl", null)
 
-      forwarded_values {
-        query_string            = lookup(i.value, "query_string", false)
-        query_string_cache_keys = lookup(i.value, "query_string_cache_keys", [])
-        headers                 = lookup(i.value, "headers", null)
-
-        cookies {
-          forward           = lookup(i.value, "cookies_forward", "none")
-          whitelisted_names = lookup(i.value, "cookies_whitelisted_names", null)
+      dynamic "forwarded_values" {
+        for_each = contains(keys(i.value), "cache_policy_id") ? [] : [{
```

Comment:
I have to admit that I didn't check the details of this specific issue but I think my comment is still relevant. :)

I believe that `for_each` values of this argument should not be dependant on whether `cache_policy_id` is set. It should be up-to the user of the module to provide `forwarded_values`. If the user wants to have `cache_policy_id` he/she should not specify any argument under the `forwarded_values` block.

So, to keep backward compatibility for existing users we need to introduce another variable `use_forwarded_values` (`default = true` for BC compatibility) and use it like:

```terraform
dynamic "forwarded_values" {
  for_each = var.use_forwarded_values ? [true] : []
  # ...
}
```

Does it make sense?

Reply:
@antonbabenko I'm just thinking that it won't work properly because `forwarded_values` depends on each item of `ordered_cache_behavior`, therefore it would need to be an option inside `ordered_cache_behaviour` and `default_cache_behavior`. Just because a user could have a mix of legacy (without cache policies, thus having forwarded_values) and new (with cache policies) cache behaviors.

So maybe it would be something like
```tf
dynamic "forwarded_values" {
  for_each = lookup(i.value, "use_forwarded_values", true) ? [true] : []
  # ...
}
```

I'll submit my changes please let me know what you think.